hardhat.config.ts

```diff
@@ -0,0 +1,56 @@
+// hardhat.config.ts
+import { HardhatUserConfig, SolcConfig } from 'hardhat/types'
+import { hardhat } from './src'
+
+import '@nomiclabs/hardhat-waffle'
+import '@typechain/hardhat'
+
+const DEFAULT_VERSION = 'v0.6'
+
+const optimizer = {
+  runs: 200,
+  enabled: true,
+}
+
+const settings = {
+  optimizer,
+  metadata: {
+    // To support Go code generation from build artifacts
+    // we need to remove the metadata from the compiled bytecode.
+    bytecodeHash: 'none',
+  },
+}
+
+const versions: Record<string, SolcConfig> = {
+  'v0.4': { version: '0.4.16', settings },
```

Comment:
This version doesn't work for me locally.
```
Downloading compiler 0.4.16
Error HH503: Couldn't download compiler version 0.4.16. Checksum verification failed. Please check your connection.
```
0.4.24 works fine though.

Reply:
Hmm, works on my machine...

`0.4.16` was the version the original LinkToken v4 was compiled with.

Reply:
Worked for me.